### ISSUE TITLE
arx fails to deal with perfect collinearity

### ISSUE THREAD

**Issue by moritzpschwarz** [gsucarrat/gets#37]
I might not have grasped this fully, but could it be that `arx` cannot deal with perfect collinearity? 

This is taken from the `arx` example and works nicely:
```r
##Simulate from an AR(1):
set.seed(123)
y <- arima.sim(list(ar=0.4), 70)

##estimate an AR(2) with intercept:
arx(y, mc=TRUE, ar=1:2)

##Simulate four independent Gaussian regressors:
xregs <- matrix(rnorm(4*70), 70, 4)

##estimate an AR(2) with intercept and four conditioning
##regressors in the mean:
arx(y, mc=TRUE, ar=1:2, mxreg=xregs)
```

But if we make a small change, e.g. adding another column that is just column 1 multiplied by 10, then we get the all too familiar error: `Error in solve.qr(qx, y, tol = tol) : singular matrix 'a' in 'solve'`

```r
xregs <- cbind(xregs,xregs[,1]*10)
arx(y, mc=TRUE, ar=1:2, mxreg=xregs)
```

Is this intentional? Should we be adding the `dropvar` function somewhere that we are already using e.g. in the course of `isat`?

**Comment by gsucarrat** [comment]
It is indeed intentional or, rather, the effort of allowing for perfect collinearity has never been made. If we are to allow for perfect collinearity in arx(), then the natural way to proceed - but this can be discussed! - is to add the argument 'singular.ok' as in lm() wtih default TRUE. Of course, internally it is natural to invoke dropvar(), but this can also be discussed. I suggest you change this thread to a 'pull request', OK?

**Comment by gsucarrat** [comment]
This issue is solved in version 0.28, where an argument similar to as in `lm()` is introduced: `singular.ok = TRUE` (default). Here is an example:
```
set.seed(123)
y <- rnorm(20)
x <- cbind(rnorm(20),2,rnorm(20))
arx(y, mc=TRUE, mxreg=x)
```
The regressor that is causing the exact colinearity is automatically removed, and a warning message about the deletion is given.
